---
layout: post
title:  "Copy number variation using HiChIP data"
date:   2024-09-08 11:54:38 -0400
categories: cnv hichip
---

## Background
Depth of coverage (DOC), off-target CNV approaches have been applied to exome and ChIP-seq data [PMC9039557][PMC4396974].
DOC-off-target methods require filtering out peak regions when using non-input immunoprecipitation (IP) samples to avoid off-target binding signals. 
This ensures that only copy number enrichment unaffected by specific binding event, are retained. 
The peak calling outcomes are highly dependent on the specific algorithm used. 
Most ChIP-seq and ATAC-seq peak callers are not optimized for detecting complex, non-symmetric peak forms. 
For instance, H3K4me3 peaks are typically sharply localized, while H3K4me1/3 peaks span broader domains. H3K27ac marks both large regions, such as super-enhancers, and smaller, discrete regions like promoters, exhibiting both broad and narrow peak characteristics [35788238][ref..].

## Problem
### Different CNVs with Different Data HiChIP(black), ChIP-seq(blue), and ChIP-seq Input (Red)
<img width="708" alt="image" src="https://github.com/user-attachments/assets/eb607ed2-29f1-409d-98ca-524c0b73a796">

### Different Peaks with Different Algorithms 
- MACS2 : Dynamic poisson parameters (λlocal = max(λBG, λ1k, λ5k, λ10k)
- HOMER : 4-fold greater than in the surrounding 10 kb region, The maximum distance used to stitch peaks together 
- CopywriteR : non-parameteric, FDR-base 
<img width="777" alt="image" src="https://github.com/user-attachments/assets/854cc534-035b-43a5-b8e7-ff66b64026e4">

## Methods 
- Modify CopywriteR algorithm (reduce expantion)
- CopywriteR Copynumber Model 
   
$$ log(C) = \beta_0( GCcontent ) + \beta_1( Mappability ) + \epsilon $$
- Code ([fullcode](https://github.com/PeeperLab/CopywriteR/blob/master/R/CopywriteR.R)
```
345                 retest.peak.ranges <- apply(test, 1, function(x) {
346                     left.lower.boundary <- max(0, (as.integer(x["start"]) - (resolution + 1)))
347                     left.higher.boundary <- max(0, (as.integer(x["start"]) - 1))
348                     right.lower.boundary <- min(chromosomes[selection],
349                                                 (as.integer(x["end"]) + 1))
350                     right.higher.boundary <- min(chromosomes[selection],
351                                                  (as.integer(x["end"]) + (resolution + 1)))
352                     left.peakCutoff <- ceiling(.peakCutoff(cov.chr[left.lower.boundary:left.higher.boundary],fdr.cutoff=FDRT))
353                     right.peakCutoff <- ceiling(.peakCutoff(cov.chr[right.lower.boundary:right.higher.boundary],fdr.cutoff=FDRT))
```

## Results
- Recovered CNV calling at 5' body of SCIRT 
<img width="1316" alt="image" src="https://github.com/user-attachments/assets/c2469821-5c32-4c47-ae33-e0ca53ae5abc">

- CNV of MG63.3 H3K27ac on Chr6 Regions
<img width="1343" alt="image" src="https://github.com/user-attachments/assets/221d2413-e53b-4706-9ad4-014c69dee9a2">


